Client/Frontend/Browser/TabManager.swift

```diff
@@ -358,27 +358,25 @@ class TabManager: NSObject {
         }
     }
 
-    //Switch between private browsing tab and normal tab
-    func switchTabMode (_ tab: Tab) -> Bool {
-        assert(Thread.isMainThread)
-        var nextTab: Tab
+    enum SwitchPrivacyModeResult { case createdNewTab; case usedExistingTab }
+    func switchPrivacyMode() -> SwitchPrivacyModeResult {
```

Comment:
This will pick the most recent tab based on lastExecutedTime from normal or private. The nice thing is this will work after an app restart also.